include/multi_blas_helper.cuh

```diff
@@ -1,23 +1,94 @@
+#pragma once
+
 #include <algorithm>
 #include <register_traits.h>
 #include <blas_helper.cuh>
+#include <kernel_helper.h>
+#include <target_device.h>
 
 namespace quda
 {
 
   namespace blas
   {
 
-    // storage for matrix coefficients
-#define MAX_MATRIX_SIZE 8192
-#define MAX_ARG_SIZE 4096
-    __constant__ signed char Amatrix_d[MAX_MATRIX_SIZE];
-    __constant__ signed char Bmatrix_d[MAX_MATRIX_SIZE];
-    __constant__ signed char Cmatrix_d[MAX_MATRIX_SIZE];
+    /**
+       @brief Returns the maximum size of a coefficient array for 2-d multi-blas
+    */
+    constexpr size_t max_array_size() { return 8192; }
```

Comment:
Should we move (and appropriately generalize) this function (and `max_arg_size` below) to target-specific backends?

Reply:
Again, this limit is algorithmic and not really target specific (at least not until we have broader sampling to more architectures).  This could be modified and made target specific if needed.